docs/dev/design/new-notifications-system.rst

```diff
@@ -0,0 +1,459 @@
+Notification system: a new approach after lot of discussions
+============================================================
+
+Notifications have been a recurrent topic in the last years.
+We have talked about different problems and solution's approaches during these years.
+However, due to the complexity of the change, and without having a clear path, it has been hard to prioritize.
+
+We've written a lot about the problems and potential solutions for the current notification system.
+This is a non-complete list of them just for reference:
+
+* https://github.com/readthedocs/readthedocs.org/issues/4226
+* https://github.com/readthedocs/readthedocs.org/issues/3399
+* https://github.com/readthedocs/meta/discussions/126
+* https://github.com/readthedocs/readthedocs.org/issues/5909
+* https://github.com/readthedocs/readthedocs.org/issues/9110
+* https://github.com/readthedocs/readthedocs.org/issues/9279
+* https://github.com/readthedocs/readthedocs.org/issues/4235
+
+At the offsite in Portland, Anthony and myself were able to talk deeply about this and wrote a bunch of thoughts in a Google Doc.
+We had pretty similar ideas and we thought we were solving most of the problems we identified already.
+
+I read all of these issues and all the discussions I found and wrote this document that summarizes my proposal:
+*create a new notification system that we can customize and expand as we need in the future*:
+
+* A Django model to store the notifications' data
+* API endpoints to retrieve the notifications for a particular resource (User, Build, Project, Organization)
+* Frontend code to display them (*outside the scope of this document*)
+
+
+Goals
+-----
+
+* Keep raising exceptions for errors from the build process
+* Ability to add non-error notifications from the build process
+* Add extra metadata associated to the notification: icon, color, header, body, etc
```

Comment:
I don't think color is super important to be sent from the backend. I added it here just in case we want to differentiate some particular notifications for some reason. In fact, if you read the proposed implementation, color is `None` and calculated based on the `type` by default: https://github.com/readthedocs/readthedocs.org/blob/776d09f72482cbc0339a9d6db11ede8d68574ebd/docs/dev/design/new-notifications-system.rst?plain=1#L130-L137

Reply:
I removed the color in 66e08b36d

Reply:
Sounds good. If we find a use case, your example for `get_display_color` is a great way to go.